src/Informer/InfoFromBuild.hpp

```diff
@@ -15,8 +15,18 @@
  * The information returned by this function is invaluable for identifying
  * the version of the code used in a simulation, as well as which host, the
  * date the code was compiled, and the time of linkage.
+ *
+ * We declare the function `extern "C"` so its symbol is not mangled and we can
```

Comment:
But it's not a C-compatible name, just as the warning suggests... This seems sketchy and very questionable. I'm not convinced this is safe

Reply:
My intention with this is to prevent the symbol being mangled so I can allow this particular symbol to remain undefined until link time on Darwin. Do you have an idea for a better way to do this? Should I try to figure out the mangled name and specify that for the `-U` option?

Reply:
Scratch that last part, name mangling can change between compiler versions..

Reply:
The only other option I can think of is allowing _any_ symbol be undefined

Reply:
You can also show me that this doesn't cause undefined behavior and that the C++ standard says it's okay :) That's what I'm worried about. I don't know what the rules are

Reply:
I couldn't find an explicit rule in the standard. The only part that comes close is this:

> Linkage from C++ to objects defined in other languages and to objects defined in C++ from other languages is implementation-defined and language-dependent. Only where the object layout strategies of two language implementations are similar enough can such linkage be achieved.

But this refers to actually linking from another language, not just staying within C++ and declaring the function `extern`.

Here's what I found people say:

> That is valid C++ code. C linkage doesn't mean the function has to actually be usable from C.
> https://gcc-help.gcc.gnu.narkive.com/DquFh2uv/warning-for-incompatible-functions-declared-extern-c

> It only impacts the name of the function in the object file by preventing name mangling. It does not make a function neither "more C" or "less C++". The only additional limitation which extern "C" adds to a C++ function is that is shall not be overloaded.
> https://stackoverflow.com/questions/4744656/functions-with-c-linkage-able-to-return-class-type

> Because the standard doesn't say this is ill-formed. The compiler is just warning you that returning a std::string is not compatible with C linkage. You'll certainly have problems if you try to call this function from C.
> https://stackoverflow.com/questions/15655912/warning-function-return-user-defined-type-which-is-incompatible-with-c

Just let me know if you prefer I change the flag so _any_ symbol can be undefined at link time, or leave as-is.

Reply:
I tracked down the origin of `-Wreturn-type-c-linkage`.  Clang copied the warning from Microsoft's [C4190](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-1-c4190), and Microsoft actually gives a rationale.  They say that it is safe as long as you neither define nor call the function in non-C++, so we're fine.

Reply:
Okay, great. Thanks for figuring that out @wthrowe!